provider/tsquery/tsquery.go

```diff
@@ -157,14 +173,168 @@ func (c cols) ToSQL() string {
 type asOfJoins []asOfJoin
 
 // ToSQL returns the SQL representation of the ASOF JOINs, space-separated.
-func (j asOfJoins) ToSQL() string {
+func (j asOfJoins) ToSQL(config QueryConfig) string {
 	joins := make([]string, len(j))
 	for i, join := range j {
 		joins[i] = join.ToSQL()
 	}
 	return strings.Join(joins, " ")
 }
 
+// asOfJoin represents an ASOF JOIN between the label and feature tables.
```

Comment:
Totally a nit, but did you mean to write _"windowJoin represents an ASOF JOIN between the label and feature tables in offline providers that don't support the ASOF operator"_?